src/casp/bq_scripts/extract_minibatch_to_anndata.py

```diff
@@ -116,17 +128,66 @@ def assign_uns_metadata(anndata, json_string):
         anndata.uns[key] = val
 
 
-def extract_minibatch_to_anndata(project, dataset, extract_table_prefix, start_bin, end_bin, output):
+def add_cell_type_info(
+    dataset: str, extract_table_prefix: str, bucket_name: str, adata: "ad.AnnData", cell_types: list
+):
+    filename = "all_cell_types.csv"
+    source_blob_name = f"{dataset}_{extract_table_prefix}_info/{filename}"
+
+    if not os.path.exists(filename):
+        utils.download_file_from_bucket(
+            bucket_name=bucket_name, source_blob_name=source_blob_name, destination_file_name=filename
+        )
+
+    adata.obs["cell_type"] = cell_types
+    df = pd.read_csv(filename, index_col=False)
+    adata.obs.cell_type = pd.Categorical(adata.obs.cell_type.values, categories=df["cell_type"].values)
+
+
+def add_expressed_genes_layer_mask(
+    dataset: str, extract_table_prefix: str, bucket_name: str, adata: "ad.AnnData", cas_ingest_ids: list
+):
+    filename = "expressed_genes_info.csv"
+    source_blob_name = f"{dataset}_{extract_table_prefix}_info/{filename}"
+
+    if not os.path.exists(filename):
+        utils.download_file_from_bucket(
+            bucket_name=bucket_name, source_blob_name=source_blob_name, destination_file_name=filename
+        )
+
+    adata.obs["cas_ingest_id"] = cas_ingest_ids
+    df = pd.read_csv(filename, index_col="cas_ingest_id").astype(bool)
+
+    expressed_genes_mask = df.loc[adata.obs["cas_ingest_id"], adata.var.index].to_numpy()
+
+    adata.layers["expressed_genes_mask"] = expressed_genes_mask
+
+
+def extract_minibatch_to_anndata(
+    project,
+    dataset,
+    extract_table_prefix,
+    start_bin,
+    end_bin,
+    output,
+    bucket_name,
+    credentials=None,
+    fq_allowed_original_feature_ids="dsp-cell-annotation-service.cas_reference_data.refdata-gex-GRCh38-2020-A",
+):
```

Comment:
Type hints and docstring. What does `fq_` in `fq_allowed_original_feature_ids` mean? Is that a typo?